Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -182,9 +182,27 @@ public class ChatChannelController: DataController, DelegateCallable, DataStoreP
     /// That is why we need to check both flag and valid `cid` before modifications.
     private var isChannelAlreadyCreated: Bool
 
-    /// A Boolean value that returns wether the previous messages have all been loaded or not.
+    /// A Boolean value that returns wether the oldest messages have all been loaded or not.
     public private(set) var hasLoadedAllPreviousMessages: Bool = false
 
+    /// A Boolean value that returns wether the newest messages have all been loaded or not.
+    public var hasLoadedAllNextMessages: Bool {
+        // We use channel.previewMessage?.createdAt and not the channel.lastMessageAt
+        // because the channel.lastMessageAt also accounts for thread replies at the moment.
+        // Right now, the previewMessage, since it is always the most recent message of the channel
+        // is a good candidate to determine if the first page is loaded (if all recent messages have been loaded).
+        channel?.previewMessage?.createdAt == messages.first?.createdAt
```

Comment:
It should be fine. But Ideally, this information would come from the server always. There is an ongoing discussion to support this since most customers don't expect `lastMessageAt` to include thread replies as well. Until then, our only option is to use the previewMessage, or create a custom property and custom logic locally to keep track of the lastMessageAt but only for channel messages. I can study this possibility as well